### ISSUE TITLE
SC 3.3.7 Accessible Authentication

### ISSUE THREAD

**Issue by ysmartin** [w3c/wcag#1318]
- The definition of "Cognitive function test" is not consistent with respect to the input of text (or, better said, the reference to cognitive functions is not consistent with some of the practical examples and techniques). More specifically, transcription of text is considered as a cognitive function, while memorization of personal identifiers (name, e-mail) is not considered as such. Ok with the definition. But in practice, the personal identifier needs to be transcribed to be used as an identifier.
- Several of the techniques provided depend on an external component (e.g. 3rd-party identity provider, 2-factor authentication). The accessibility of such mechanisms (which may go beyond Web technologies) needs to be ensured.
- Likewise, it should be emphasized that the email sent as per [G218: Email link authentication](https://www.w3.org/WAI/WCAG22/Techniques/general/G218) should also be accessible. For instance, it should not require further cognitive effort. In particular, the usual sequence implied by this task would be as follows:
  1) switching from the browser or web application to the e-mail client,
  2) waiting for the incoming e-mail,
  3) clicking on the link, 
  4) which switches back to the browser (ideally, as on some platforms, the website is open within the e-mail client).

  The changes of context implied by this sequence can prove difficult for people with cognitive limitations addressed by SC 3.3.7 itself.
- Other authentication mechanisms which are not yet mentioned may exist and do not involve interaction from the user (based on e.g. device location, device identification, user-agent storage e.g. cookies, etc.) Even if the application of these other mechanisms does not fit every situation, they should not be discarded if they are accessible.

**Comment by lgnacio** [comment]
- According to [G218: Email link authentication](https://www.w3.org/WAI/WCAG22/Techniques/general/G218), there is a issues where you need to login to e-mail from another website for Email link authentication. 

- E-mail login of "B website" is required for Email link authentication of "A website". Then, to use Email link authentication, the premise that "B website" must comply with 3.3.7 is required.

**Comment by alastc** [comment]
> More specifically, transcription of text is considered as a cognitive function, while memorization of personal identifiers (name, e-mail) is not considered as such. 

Text specific to a website (e.g. username, password) varies by website, and each user can end up with 100s of these. (E.g. I have over 700 logins in my password manager.) Whereas your personal identifiers do not change per website. For another issue we've [adjusted the definition](https://github.com/w3c/wcag/pull/1411/files) to make that clearer.

> Several of the techniques provided depend on an external component (e.g. 3rd-party identity provider, 2-factor authentication). The accessibility of such mechanisms (which may go beyond Web technologies) needs to be ensured.
In the context of WCAG, we can only ask that the author of the website making a conformance claim meets the guidelines. 

If the mechanism is direct we can reference 'accessibility supported', that's built in to the conformance model. Whether a 3rd party website/feature is accessibility is a separate problem, probably best dealt with at the regulation/legal level.
Also, if one of the recommended solutions is chosen (like webauthn), the user has flexibility as to which mechanism they use. I.e. it is not defined by the website, but the user.

> The changes of context implied by this sequence can prove difficult for people with cognitive limitations addressed by SC 3.3.7 itself.

I assume that is true, but according to the Cognitive & Learning Disabilities Task Force, this approach is preferable to typing in usernames & passwords.

> Other authentication mechanisms which are not yet mentioned may exist … they should not be discarded if they are accessible.

The techniques referenced are known methods, it is true of every criteria that you can use other methods so long as they meet the SC text.

**Comment by ysmartin** [comment]
> > More specifically, transcription of text is considered as a cognitive function, while memorization of personal identifiers (name, e-mail) is not considered as such.
> 
> Text specific to a website (e.g. username, password) varies by website, and each user can end up with 100s of these. (E.g. I have over 700 logins in my password manager.) Whereas your personal identifiers do not change per website. For another issue we've [adjusted the definition](https://github.com/w3c/wcag/pull/1411/files) to make that clearer.
> 
Much clearer now within the definition itself.

I have realized a secondary issue regarding this use of a single e-mail as an identifier across websites: the reuse of the same identifier (in this case, same e-mail address) may raise potential privacy issues (specifically, linkability across user accounts). I would suggest consulting with @w3cping on that topic.

**Comment by alastc** [comment]
We aren't trying to steer people to using emails as such, but we will undergo a horizontal review. I know that includes security, but I'll check about the privacy group.

Did the rest of the response make sense, are you happy to close the issue?

**Comment by alastc** [comment]
Draft group response:
<hr>
With the changes linked to above, and the comments in the thread have resolved / answered the issues you raised.

**Comment by alastc** [comment]
The group agreed with the response above, please re-open this issue if something has not been addressed.
